### PR TITLE
Feat/oauth2 jwt consumer

### PR DESCRIPTION
**Pull request by leekeal**
Can you consider the default consumerId, Instead of using headers policy?

### CODE REVIEW

**Comment by XVincentX** [comment]
Thanks for the PR.

Can you explain a little bit more your use case? This is already handled in another PR and I do not really get why that line of code to be honest.

File: lib/policies/jwt/jwt.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ module.exports = function (params) {
     }
 
     if (!params.checkCredentialExistence) {
-      return done(null, { id: 'anonymous' });
+      jwtPayload.consumerId && (jwtPayload.id = jwtPayload.consumerId);
```
Comment (XVincentX) [inline lib/policies/jwt/jwt.js]: I think this is wrong.

The use case here is that, if we want to bypass the express-gateway identity system, we should just forward the payload and not make any assumption on it. Who's telling you that `jwtPayload.id` is always valorized and that its value corresponds to the consumer Id? That depends on your token provider.

Also, you're modifying the payload, while I think we should not touch it at all.

**Comment by XVincentX** [comment]
I'll close this until @leekeal will provide further informations.

File: lib/policies/jwt/jwt.js (hunk 1 above)
Comment (leekeal) [reply lib/policies/jwt/jwt.js]: Understand what you mean.
I think you are right.
thank you.

**Pull request closed without merging**